include/mbgl/gfx/dynamic_texture_atlas.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <mbgl/gfx/dynamic_texture.hpp>
+#include <mbgl/text/glyph.hpp>
+#include <mbgl/style/image_impl.hpp>
+
+namespace mbgl {
+
+namespace gfx {
+
+using DynamicTexturePtr = std::shared_ptr<gfx::DynamicTexture>;
+
+class GlyphAtlas {
+public:
+    GlyphPositions glyphPositions;
+    std::vector<TextureHandle> textureHandles;
+    DynamicTexturePtr dynamicTexture;
+};
+
+class ImageAtlas {
+public:
+    ImagePositions iconPositions;
+    ImagePositions patternPositions;
+    std::vector<TextureHandle> textureHandles;
+    DynamicTexturePtr dynamicTexture;
+};
+
+class DynamicTextureAtlas {
+public:
+    DynamicTextureAtlas(Context& context_)
+        : context(context_) {}
+    ~DynamicTextureAtlas() = default;
+
+    GlyphAtlas uploadGlyphs(const GlyphMap& glyphs);
+    ImageAtlas uploadIconsAndPatterns(const ImageMap& icons,
+                                      const ImageMap& patterns,
+                                      const ImageVersionMap& versionMap);
+
+    void removeTextures(const std::vector<TextureHandle>& textureHandles, const DynamicTexturePtr& dynamicTexture);
+
+private:
+    Context& context;
+    std::vector<DynamicTexturePtr> dynamicTextures;
```

Comment:
Is it possible for dynamic textures outlive the glyph atlas?

Reply:
Yes, for short time when the style is changed.